diff --git a/src/bin/main.c b/src/bin/main.c
index 479515f..3c8c980 100644
--- a/src/bin/main.c
+++ b/src/bin/main.c
@@ -326,9 +326,7 @@ static const Ecore_Getopt options = {
    "Terminal emulator written with Enlightenment Foundation Libraries.",
    EINA_TRUE,
    {
-      ECORE_GETOPT_BREAK_STR ('e', "exec",
-                              "command to execute. "
-                              "Defaults to $SHELL (or passwd shel or /bin/sh)"),
+
       ECORE_GETOPT_STORE_STR ('d', "current-directory",
                               "Change to directory for execution of terminal command."),
       ECORE_GETOPT_STORE_STR ('t', "theme",
@@ -406,9 +404,7 @@ elm_main(int argc, char **argv)
    Eina_Bool maximized = EINA_FALSE;
    Eina_Bool nowm = EINA_FALSE;
    Eina_Bool quit_option = EINA_FALSE;
-   Eina_Bool cmd_options = EINA_FALSE;
-   Ecore_Getopt_Value values[] = {
-     ECORE_GETOPT_VALUE_BOOL(cmd_options),
+    Ecore_Getopt_Value values[] = {
      ECORE_GETOPT_VALUE_STR(cd),
      ECORE_GETOPT_VALUE_STR(theme),
      ECORE_GETOPT_VALUE_STR(background),
@@ -473,27 +469,7 @@ elm_main(int argc, char **argv)
 
    if (quit_option) goto end;
 
-   if (cmd_options)
-     {
-        int i;
-        Eina_Strbuf *strb;
 
-        if (args == argc)
-          {
-             fprintf(stdout, "ERROR: option %s requires an argument!\n", argv[args-1]);
-             fprintf(stdout, "ERROR: invalid options found. See --help.\n");
-             goto end;
-          }
-        
-        strb = eina_strbuf_new();
-        for(i = args; i < argc; i++)
-          {
-             eina_strbuf_append(strb, argv[i]);
-             eina_strbuf_append_char(strb, ' ');
-          }
-        cmd = eina_strbuf_string_steal(strb);
-        eina_strbuf_free(strb);
-     }
 
    if (theme)
      {
